pcdsdevices/targets.py

```diff
@@ -260,3 +267,795 @@ def move(self, nxspaces, nyspaces, wait=False):
+        Get the information for a saved sample.
+
+        Parameters
+        ----------
+        sample_name : str
+            The sample name that we want the grid for. To see current
+            available samples call `mapped_grids`
+        path : str, optional
+            Path to the `.yml` file. Defaults to the path defined when
+            creating this object.
+
+        Returns
+        -------
+        data : dictionary
+            Dictionary of all the information for a saved sample.
+
+        Examples
+        --------
+        >>> get_sample('sample1')
+        {'time_created': '2021-01-06 11:43:40.701095',
+        'top_left': [0, 0],
+        'top_right': [4.0, -1.0],
+        'bottom_right': [4.4, -3.5],
+        'bottom_left': [1.0, -3.0],
+        'M': 10,
+        'N': 10,
+        'coefficients': [1.1686746987951824,
+        -0.3855421686746996,
+        -9.730859023513261e-15,
+        -0.29216867469879476,
+        1.1566265060240974,
+        6.281563288265657e-16,
+        0.042168674698794054,
+        -0.05220883534136586]}
+        """
+        path = path or self._path
+        data = None
+        with open(path) as sample_file:
+            try:
+                data = yaml.safe_load(sample_file)
+            except yaml.YAMLError as err:
+                logger.error('Error when loading the samples yaml file: %s',
+                             err)
+        if data is None:
+            logger.warning('The file is empty, no sample grid yet. '
+                           'Please use `save_presets` to insert grids '
+                           'in the file.')
+        try:
+            return data[str(sample_name)]
+        except Exception:
+            logger.error('The sample %s might not exist in the file.',
+                         sample_name)
+
+    def get_sample_map_info(self, sample_name, path=None):
+        """
+        Given a sample name, get the m and n points, as well as the coeffs.
+
+        Parameters
+        ----------
+        sample_name : str
+            The name of the sample to get the mapped points from. To see the
+            available mapped samples call the `mapped_samples()` method.
+        path : str, optional
+            Path to the samples yaml file.
+        """
+        path = path or self._path
+        sample = self.get_sample_data(str(sample_name))
+        coeffs = []
+        m_points, n_points = 0, 0
+        if sample:
+            try:
+                coeffs = sample["coefficients"]
+                m_points = sample['M']
+                n_points = sample['N']
+            except Exception as ex:
+                logger.error('Something went wrong when getting the '
+                             'information for sample %s. %s', sample_name, ex)
+                return
+        else:
+            logger.error('This sample probably does not exist. Please call'
+                         ' mapped_samples() to see which ones are available.')
+            return
```

Comment:
Another one where the ambiguous return type should be documented or made non-ambiguous

Reply:
thanks - fixed this in 393ee51